### PR TITLE
cpu/numactl: Add checks for missing parameters

### PR DESCRIPTION
**Pull request by sacsant**
commit 611b2ed added support for Network and FC devices. While doing so
it changed the default behaviour of existing test. If no yaml input is provided
the test now cancels all sub-tests.

Introduce checks for various yaml parameters and skip corresponding tests
if one or more required paramter is missing.

Signed-off-by: Sachin Sant <sachinp@linux.ibm.com>

### CODE REVIEW

**Comment by sacsant** [comment]
Before fix:

(1/7) cpu/numactl.py:Numactl.test: STARTED
 (1/7) cpu/numactl.py:Numactl.test: CANCEL: PCI_address not given (7.29 s)
 (2/7) cpu/numactl.py:Numactl.test_interleave: STARTED
 (2/7) cpu/numactl.py:Numactl.test_interleave: CANCEL: PCI_address not given (7.50 s)
 (3/7) cpu/numactl.py:Numactl.test_localalloc: STARTED
 (3/7) cpu/numactl.py:Numactl.test_localalloc: CANCEL: PCI_address not given (7.32 s)
 (4/7) cpu/numactl.py:Numactl.test_preferred_node: STARTED
 (4/7) cpu/numactl.py:Numactl.test_preferred_node: CANCEL: PCI_address not given (7.60 s)
 (5/7) cpu/numactl.py:Numactl.test_cpunode_with_membind: STARTED
 (5/7) cpu/numactl.py:Numactl.test_cpunode_with_membind: CANCEL: PCI_address not given (7.26 s)
 (6/7) cpu/numactl.py:Numactl.test_physical_cpu_bind: STARTED
 (6/7) cpu/numactl.py:Numactl.test_physical_cpu_bind: CANCEL: PCI_address not given (7.49 s)
 (7/7) cpu/numactl.py:Numactl.test_numa_pci_bind: STARTED
 (7/7) cpu/numactl.py:Numactl.test_numa_pci_bind: CANCEL: PCI_address not given (7.45 s)

After fix:

1/7) cpu/numactl.py:Numactl.test: STARTED
 (1/7) cpu/numactl.py:Numactl.test: WARN: Test passed but there were warnings during execution. Check the log for details. (9.41 s)
 (2/7) cpu/numactl.py:Numactl.test_interleave: STARTED
 (2/7) cpu/numactl.py:Numactl.test_interleave: CANCEL: Network inferface or disk/device input missing (7.33 s)
 (3/7) cpu/numactl.py:Numactl.test_localalloc: STARTED
 (3/7) cpu/numactl.py:Numactl.test_localalloc: CANCEL: Network inferface or disk/device input missing (7.30 s)
 (4/7) cpu/numactl.py:Numactl.test_preferred_node: STARTED
 (4/7) cpu/numactl.py:Numactl.test_preferred_node: CANCEL: Network inferface or disk/device input missing (7.38 s)
 (5/7) cpu/numactl.py:Numactl.test_cpunode_with_membind: STARTED
 (5/7) cpu/numactl.py:Numactl.test_cpunode_with_membind: CANCEL: Network inferface or disk/device input missing (7.31 s)
 (6/7) cpu/numactl.py:Numactl.test_physical_cpu_bind: STARTED
 (6/7) cpu/numactl.py:Numactl.test_physical_cpu_bind: CANCEL: Network inferface or disk/device input missing (7.43 s)
 (7/7) cpu/numactl.py:Numactl.test_numa_pci_bind: STARTED
 (7/7) cpu/numactl.py:Numactl.test_numa_pci_bind: CANCEL: Device input missing, skipping the test (7.43 s)

**Comment by sacsant** [comment]
Thanks @FarooqAbdulla01 for the test feedback. I have pushed an updated version to fix the issues found during testing.

**Comment by sacsant** [comment]
avocado run --nrunner-max-parallel-tasks=1 cpu/numactl.py
 (1/7) cpu/numactl.py:Numactl.test: STARTED
 (1/7) cpu/numactl.py:Numactl.test: PASS (133.52 s)
 (2/7) cpu/numactl.py:Numactl.test_interleave: STARTED
 (2/7) cpu/numactl.py:Numactl.test_interleave: CANCEL: Network inferface or disk/device input missing (7.89 s)
 (3/7) cpu/numactl.py:Numactl.test_localalloc: STARTED
 (3/7) cpu/numactl.py:Numactl.test_localalloc: CANCEL: Network inferface or disk/device input missing (8.06 s)
 (4/7) cpu/numactl.py:Numactl.test_preferred_node: STARTED
 (4/7) cpu/numactl.py:Numactl.test_preferred_node: CANCEL: Network inferface or disk/device input missing (7.81 s)
 (5/7) cpu/numactl.py:Numactl.test_cpunode_with_membind: STARTED
 (5/7) cpu/numactl.py:Numactl.test_cpunode_with_membind: CANCEL: Network inferface or disk/device input missing (7.79 s)
 (6/7) cpu/numactl.py:Numactl.test_physical_cpu_bind: STARTED
 (6/7) cpu/numactl.py:Numactl.test_physical_cpu_bind: CANCEL: Network inferface or disk/device input missing (8.14 s)
 (7/7) cpu/numactl.py:Numactl.test_numa_pci_bind: STARTED
 (7/7) cpu/numactl.py:Numactl.test_numa_pci_bind: CANCEL: Device input missing, skipping the test (7.85 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 6

avocado run --nrunner-max-parallel-tasks=1 cpu/numactl.py -m cpu/numactl.py.data/numa_network.yaml 
 (1/7) cpu/numactl.py:Numactl.test;run-88e5: STARTED
 (1/7) cpu/numactl.py:Numactl.test;run-88e5: PASS (137.72 s)
 (2/7) cpu/numactl.py:Numactl.test_interleave;run-88e5: STARTED
 (2/7) cpu/numactl.py:Numactl.test_interleave;run-88e5: PASS (12.00 s)
 (3/7) cpu/numactl.py:Numactl.test_localalloc;run-88e5: STARTED
 (3/7) cpu/numactl.py:Numactl.test_localalloc;run-88e5: PASS (12.16 s)
 (4/7) cpu/numactl.py:Numactl.test_preferred_node;run-88e5: STARTED
 (4/7) cpu/numactl.py:Numactl.test_preferred_node;run-88e5: PASS (12.31 s)
 (5/7) cpu/numactl.py:Numactl.test_cpunode_with_membind;run-88e5: STARTED
 (5/7) cpu/numactl.py:Numactl.test_cpunode_with_membind;run-88e5: PASS (12.04 s)
 (6/7) cpu/numactl.py:Numactl.test_physical_cpu_bind;run-88e5: STARTED
 (6/7) cpu/numactl.py:Numactl.test_physical_cpu_bind;run-88e5: PASS (12.06 s)
 (7/7) cpu/numactl.py:Numactl.test_numa_pci_bind;run-88e5: STARTED
 (7/7) cpu/numactl.py:Numactl.test_numa_pci_bind;run-88e5: PASS (12.28 s)
RESULTS    : PASS 7 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

avocado run --nrunner-max-parallel-tasks=1 cpu/numactl.py -m cpu/numactl.py.data/numa_fc.yaml      
 (1/7) cpu/numactl.py:Numactl.test;run-bytes-count-input_device-seek-a7f3: STARTED
 (1/7) cpu/numactl.py:Numactl.test;run-bytes-count-input_device-seek-a7f3: PASS (133.52 s)
 (2/7) cpu/numactl.py:Numactl.test_interleave;run-bytes-count-input_device-seek-a7f3: STARTED
 (2/7) cpu/numactl.py:Numactl.test_interleave;run-bytes-count-input_device-seek-a7f3: PASS (12.22 s)
 (3/7) cpu/numactl.py:Numactl.test_localalloc;run-bytes-count-input_device-seek-a7f3: STARTED
 (3/7) cpu/numactl.py:Numactl.test_localalloc;run-bytes-count-input_device-seek-a7f3: PASS (12.27 s)
 (4/7) cpu/numactl.py:Numactl.test_preferred_node;run-bytes-count-input_device-seek-a7f3: STARTED
 (4/7) cpu/numactl.py:Numactl.test_preferred_node;run-bytes-count-input_device-seek-a7f3: PASS (12.54 s)
 (5/7) cpu/numactl.py:Numactl.test_cpunode_with_membind;run-bytes-count-input_device-seek-a7f3: STARTED
 (5/7) cpu/numactl.py:Numactl.test_cpunode_with_membind;run-bytes-count-input_device-seek-a7f3: PASS (12.78 s)
 (6/7) cpu/numactl.py:Numactl.test_physical_cpu_bind;run-bytes-count-input_device-seek-a7f3: STARTED
 (6/7) cpu/numactl.py:Numactl.test_physical_cpu_bind;run-bytes-count-input_device-seek-a7f3: PASS (12.24 s)
 (7/7) cpu/numactl.py:Numactl.test_numa_pci_bind;run-bytes-count-input_device-seek-a7f3: STARTED
 (7/7) cpu/numactl.py:Numactl.test_numa_pci_bind;run-bytes-count-input_device-seek-a7f3: PASS (8.00 s)
RESULTS    : PASS 7 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

File: cpu/numactl.py [hunk 1]
```diff
@@ -193,6 +194,8 @@ def test_localalloc(self):
         '''
         Test memory allocation on the current node
         '''
+        if not self.iface and not self.disk:
```
Comment (abdhaleegit) [inline cpu/numactl.py:197]: Should this be OR instead and as at any point I will be running either NIC NUma woth only interface input or FC NUMA with disk as input

Can this be taken care in setup instead in test ?

Comment (sacsant) [reply cpu/numactl.py:197]: The condition for cancel is if both iface and disk values are not specified by user. It cannot be OR as all the tests (except first) would then be cancelled.
If the user provides any one or both the values then the corresponding tests will be executed.

I cannot add this check in setup as that would cancel first test also (which was the original problem this fix addresses)

**Review by abdhaleegit — APPROVED** [review]
LGTM, Thank you @sacsant  for the patch

**Pull request merged**